<launch>

  <arg name="rviz" default="true" />

  <node pkg="nodelet" type="nodelet" name="nodelet_manager" args="manager"/>

    <!-- Sync RGB/Depth images per camera -->
    <node pkg="nodelet" type="nodelet" name="rgbd_sync1" args="load rtabmap_sync/rgbd_sync nodelet_manager" respawn="true" output = "screen">
        <remap from="rgb/image" to="/realsense/rgb/image_raw"/>
        <remap from="depth/image" to="/realsense/aligned_depth_to_color/image_raw"/>
        <remap from="rgb/camera_info" to="/realsense/rgb/camera_info"/>
        <remap from="rgbd_image" to="rgbd_image"/>
        <param name="approx" value="true"/>
        <param name="queue_size" value="50"/>
    </node>

  <!-- RTAB-Map node -->
  <node name="rtabmap" pkg="rtabmap_slam" type="rtabmap" output="screen" args="--delete_db_on_start">
    <param name="subscribe_depth"  type="bool"   value="false"/>
    <param name="subscribe_rgbd"   type="bool"   value="true"/>
    <param name="subscribe_rgb"    type="bool"   value="false"/>
    <param name="gen_scan"         type="bool"   value="true"/>
    <param name="subscribe_scan"   type="bool"   value="true"/>
    <param name="frame_id"         type="string" value="mir/base_link"/>
    <param name="odom_frame_id"    type="string" value="mir/base_link"/>  
    <param name="publish_tf"       type="bool"   value="true" />
    <param name="tf_prefix"        type="string" value="rtabmap" />
    <param name="map_frame_id"     type="string" value="map"/>
    

    <remap from="scan" to="/scan"/>
    <remap from="rgbd_image" to="rgbd_image"/>
    <remap from="odom" to="/noodom"/>
    
    <param name="Grid/FromDepth"     type="string" value="true"/>
    <param name="Vis/EstimationType" type="string" value="1"/> <!-- should be 0 for multi-cameras -->

    <!-- RTAB-Map's parameters -->
    <param name="Rtabmap/DetectionRate" type="int" value = "5"/>
  </node>

  <node if="$(arg rviz)" pkg="rviz" type="rviz" name="rviz"/>
</launch>
